packages/blade/src/components/Collapsible/CollapsibleChevronIcon.native.tsx

```diff
@@ -0,0 +1,39 @@
+import Animated, { useAnimatedStyle, useDerivedValue, withTiming } from 'react-native-reanimated';
+import { useCollapsible } from './CollapsibleContext';
+import {
+  getCollapsibleChevronIconTransforms,
+  getTransitionDuration,
+  getTransitionEasing,
+} from './commonStyles';
+import { castNativeType, makeAccessible } from '~utils';
+import type { IconComponent } from '~components/Icons';
+import { ChevronDownIcon } from '~components/Icons';
+import { useTheme } from '~components/BladeProvider';
+
+// Not really an IconComponent, a wrapper is needed for animating the icon inside
+const CollapsibleChevronIcon: IconComponent = (props) => {
+  const { isExpanded, direction } = useCollapsible();
+  const { theme } = useTheme();
+
+  const { transformExpanded, transformCollapsed } = getCollapsibleChevronIconTransforms({
+    direction,
+  });
+  const duration = castNativeType(getTransitionDuration(theme));
+  const easing = castNativeType(getTransitionEasing(theme));
+
+  const rotateZ = useDerivedValue(() =>
+    withTiming(isExpanded ? transformExpanded : transformCollapsed, { duration, easing }),
+  );
+  const animatedStyles = useAnimatedStyle(() => ({
+    // Interpolated string has to be passed, to make it work for both android and ios
+    transform: [{ rotateZ: `${rotateZ.value}deg` }],
```

Comment:
interesting, this is documented somewhere on the Reanimated docs? or did you stumble upon it randomly?

Reply:
Random searches and lot of pulling my hair 💀

Reply:
Didn't get this? Anything different here? We're doing same for Spinner as well, right?

Reply:
I think I might have been passing `rotateZ.value` directly earlier, lemme get rid of the comment. Weirdly didn't see any TS errors that time but now it's correctly throwing TS errors lol

Reply:
Getting rid of the comment in future PR